ui/components/app/confirm/info/row/url.tsx

```diff
@@ -16,13 +16,32 @@ import {
   TextVariant,
   BackgroundColor,
 } from '../../../../../helpers/constants/design-system';
+import SnapAuthorshipPill from '../../../snaps/snap-authorship-pill';
+import { SnapMetadataModal } from '../../../snaps/snap-metadata-modal';
 
 export type ConfirmInfoRowUrlProps = {
   url: string;
 };
 
 export const ConfirmInfoRowUrl = ({ url }: ConfirmInfoRowUrlProps) => {
   let urlObject;
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const openModal = () => setIsModalOpen(true);
+  const closeModal = () => setIsModalOpen(false);
+  const isSnapId = url.startsWith('npm:') || url.startsWith('local:');
```

Comment:
Can we introduce/use the function from https://github.com/MetaMask/metamask-extension/pull/26715

Reply:
Yeah, I was thinking about that. I guess we can wait to merge that first.

Reply:
[cf7c078](https://github.com/MetaMask/metamask-extension/pull/26881/commits/cf7c0781b3527c7b71c50e4402398a85c6f8e595)